src/main/kotlin/org/rust/lang/core/completion/RsReprCompletionProvider.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.completion
+
+import com.intellij.codeInsight.completion.CompletionParameters
+import com.intellij.codeInsight.completion.CompletionResultSet
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import com.intellij.patterns.ElementPattern
+import com.intellij.patterns.PlatformPatterns
+import com.intellij.psi.PsiElement
+import com.intellij.util.ProcessingContext
+import org.rust.lang.RsLanguage
+import org.rust.lang.core.psi.RsEnumItem
+import org.rust.lang.core.psi.RsMetaItem
+import org.rust.lang.core.psi.RsPath
+import org.rust.lang.core.psi.RsStructItem
+import org.rust.lang.core.psi.ext.RsStructOrEnumItemElement
+import org.rust.lang.core.psi.ext.ancestorStrict
+import org.rust.lang.core.psiElement
+import org.rust.lang.core.types.ty.TyInteger
+import org.rust.lang.core.withSuperParent
+
+object RsReprCompletionProvider : RsCompletionProvider() {
+    override val elementPattern: ElementPattern<out PsiElement>
+        get() = PlatformPatterns.psiElement()
+            .withLanguage(RsLanguage)
+            .withParent(psiElement<RsPath>()
+                .withParent(psiElement<RsMetaItem>()
+                    .withSuperParent(2,
+                        psiElement<PsiElement>()
+                            .withSuperParent<RsStructOrEnumItemElement>(2)
+                    )
+                )
+            )
+
+    override fun addCompletions(parameters: CompletionParameters, context: ProcessingContext, result: CompletionResultSet) {
+        val owner = parameters.position.ancestorStrict<RsStructOrEnumItemElement>()
+            ?: return
+
+        fun Sequence<String>.addToResults() = forEach {
+            val element = LookupElementBuilder.create(it)
+            if (it.endsWith("()")) {
+                // TODO: Caret must land in between parentheses.
```

Comment:
I'm not sure how to do this.

Reply:
You can try to use `.withInsertHandler` and create an insert handler that will move the caret after the item is inserted.